addons/dockable_container/dockable_container.gd

```diff
@@ -128,11 +129,18 @@ func _can_drop_data(_position: Vector2, data) -> bool:
 
 
 func _drop_data(_position: Vector2, data) -> void:
-	var from_node := get_node(data.from_path) as DockablePanel
+	var from_node: DockablePanel
+	var tab_index: int
+	if _godot_version.major <= 4 and _godot_version.minor <= 1:
+		from_node = get_node(data.from_path)
+		tab_index = data.tabc_element
+	else:  # Godot 4.2 and on
+		from_node = get_node(data.from_path).get_parent()
+		tab_index = data.tab_index
```

Comment:
Instead of checking for Godot version, we could try 4.2 style first, than fallback to 4.1.

For `from_node`, we can get the value, then go with something like `if from_node is TabBar: from_node = from_node.get_parent()` or `if not from_node is TabContainer: from_node = from_node.get_parent()`.

For `tab_index`, which can go back to being fetched after the early return, can try getting "tab_index" first, than falling back to "tabc_element".